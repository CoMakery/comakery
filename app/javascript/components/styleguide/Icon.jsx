import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'

class Icon extends React.Component {
  render() {
    const {className, name, ...other} = this.props

    const classnames = classNames(
      'icon',
      `icon__${name.replace('.', '-')}`,
      className
    )

    return (
      <React.Fragment>
        <img className={classnames} {...other}
          src={require(`src/images/styleguide/icons/${name}`)}
        />
      </React.Fragment>
    )
  }
}

Icon.propTypes = {
  className: PropTypes.string,
  name     : PropTypes.string
}
Icon.defaultProps = {
  className: '',
  name     : 'atomsIconsSystemHeart.svg'
}
export default Icon
