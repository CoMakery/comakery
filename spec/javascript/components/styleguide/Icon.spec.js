import React from 'react'
import { shallow, mount, render } from 'enzyme'
import Icon from 'components/styleguide/Icon'

describe('Icon', () => {
  it('renders correctly without props', () => {
    const wrapper = shallow(<Icon />)

    expect(wrapper).toMatchSnapshot()
    expect(wrapper.exists('.icon')).toBe(true)
    expect(wrapper.exists('.icon__Atoms---Icons-System-Heart-png')).toBe(true)
  })

  it('renders correctly with custom className', () => {
    const wrapper = shallow(<Icon className="__test" />)

    expect(wrapper.exists('.icon.__test')).toBe(true)
  })

  it('renders correctly with custom name', () => {
    const wrapper = shallow(<Icon name="ICON-DONE.png" />)

    expect(wrapper.exists('.icon__ICON-DONE-png')).toBe(true)
  })
})
