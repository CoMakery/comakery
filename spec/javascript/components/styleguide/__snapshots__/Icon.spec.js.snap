// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon renders correctly without props 1`] = `
<Fragment>
  <img
    className="icon icon__atomsIconsSystemHeart-svg"
    src={null}
  />
</Fragment>
`;
